tfhe-zk-pok/src/proofs/pke_v2.rs

```diff
@@ -2357,165 +2357,64 @@ pub fn verify<G: Curve>(
 
 #[cfg(test)]
 mod tests {
+    use super::super::test::*;
     use super::*;
-    use bincode::ErrorKind;
     use rand::rngs::StdRng;
     use rand::{Rng, SeedableRng};
 
-    fn polymul_rev(a: &[i64], b: &[i64]) -> Vec<i64> {
-        assert_eq!(a.len(), b.len());
-        let d = a.len();
-        let mut c = vec![0i64; d];
-
-        for i in 0..d {
-            for j in 0..d {
-                if i + j < d {
-                    c[i + j] = c[i + j].wrapping_add(a[i].wrapping_mul(b[d - j - 1]));
-                } else {
-                    c[i + j - d] = c[i + j - d].wrapping_sub(a[i].wrapping_mul(b[d - j - 1]));
-                }
-            }
-        }
-
-        c
-    }
-
     #[test]
     fn test_pke() {
         let d = 2048;
         let k = 320;
-        let B = 1048576;
+        let b_i = 1048576;
         let q = 0;
         let t = 1024;
         let msbs_zero_padding_bit_count = 1;
         let effective_cleartext_t = t >> msbs_zero_padding_bit_count;
 
-        let delta = {
-            let q = if q == 0 { 1i128 << 64 } else { q as i128 };
-            // delta takes the encoding with the padding bit
-            (q / t as i128) as u64
-        };
-
         let rng = &mut StdRng::seed_from_u64(0);
 
-        let a = (0..d).map(|_| rng.gen::<i64>()).collect::<Vec<_>>();
-        let s = (0..d)
-            .map(|_| (rng.gen::<u64>() % 2) as i64)
-            .collect::<Vec<_>>();
-        let e = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
-            .collect::<Vec<_>>();
-        let e1 = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
-            .collect::<Vec<_>>();
+        let PkeTestParams {
+            a,
+            e1,
+            e2,
+            r,
+            m,
+            b,
+            c1,
+            c2,
+            metadata,
+        } = PkeTestParams::gen(rng, d, k, b_i, q, t, msbs_zero_padding_bit_count);
+
         let fake_e1 = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
-            .collect::<Vec<_>>();
-        let e2 = (0..k)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
+            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
             .collect::<Vec<_>>();
         let fake_e2 = (0..k)
-            .map(|_| (rng.gen::<u64>() % (2 * B)) as i64 - B as i64)
+            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
```

Comment:
are we sure about the B vs b_i ?

Reply:
in both proofs they play the same role (noise bound), so I think we should give them the same name in pke and pke_v2. Before the pr this is called B in pke and b_i in pke_v2.

Reply:
I think it's linked to the notations in the technical notes, they are not strictly the same IIRC as we need to derive an upper bound of the std dev in v2

Reply:
I used B everywhere and used the value from the TUniform params